reason_4.0/lib/core/minisite_templates/modules/events.php

```diff
@@ -3719,6 +3769,25 @@ function get_ical($events)
   		return $calendar -> get_icalendar_events();
 	}
 	/**
+	 * Get a json representation of a set of events
+	 * @param array $events entities
+	 * @return string JSON
+	 */
+	function get_json($events)
+	{
+		if(!is_array($events))
+		{
+			trigger_error('get_json needs an array of event entities');
+			return '';
+		}
+
+		$events = array_values(array_map(function($e) {
+			return $e->get_values();
+		}, $events));
+
+		return json_encode($events);
```

Comment:
If you get data that is not encoded properly, `json_encode()` will crash. There is some code in the `4.8` branch I added recently that handles this case for json feeds.

The issue is an artifact of data that was inserted under odd circumstances (collation of the DB and data itself were not aligned.) In odd cases the data is UTF-8 but the database is not setup properly for that.

I added some code here for feeds: https://github.com/reasoncms/reasoncms/blob/4.8/reason_4.0/lib/core/classes/api/feed/models/reason_json.php#L228-L253

Those functions could be extracted out into a library or helper so that they could be used here.